renderdoc/maths/formatpacking.cpp

```diff
@@ -595,97 +565,160 @@ FloatVector DecodeFormattedComponents(const ResourceFormat &fmt, const byte *dat
       comp++;
       data += fmt.compByteWidth;
     }
-
-    if(fmt.type == ResourceFormatType::A8)
-    {
-      ret.w = ret.x;
-      ret.x = 0.0f;
-    }
-    else if(fmt.type == ResourceFormatType::S8)
-    {
-      ret.y = ret.x;
-      ret.x = 0.0f;
-    }
-
-    if(fmt.BGRAOrder())
-      std::swap(ret.x, ret.z);
   }
   else
   {
     if(success)
       *success = false;
   }
-
-  return ret;
 }
 
-void EncodeFormattedComponents(const ResourceFormat &fmt, FloatVector v, byte *data, bool *success)
+void DecodePixelData(const ResourceFormat &fmt, const byte *data, PixelValue &out, bool *success)
 {
-  uint64_t dummy[4] = {};
+  out.floatValue[0] = 0.0f;
```

Comment:
All this writing of individual components can be prone to copy/paste errors, would it be worth adding some helpers to do the casting and assigning? It may make the encode/decode functions more readable too.